### ISSUE TITLE
Add support for C18 standard

### ISSUE THREAD

**Issue by gerardoti** [Azure/azure-iot-sdk-c#1083]
TI CC32XX or MSP432E4 devices
IAR Embedded Workbench for ARM 8.40.1

Release lts_01_2019 (54dd89ccd004f72ab59ab1687bb9aa56ba37138c)

If the Azure C SDK is compiled with a toolchain using the C18 standard revision there are errors thrown because of various checks checking the C standard revision is either C99 or C11. These should also support C18. 
This:
```
#if ((__STDC_VERSION__  == 199901L) || (__STDC_VERSION__ == 201112L))
/*C99 compiler or C11*/
```
Should be changed to include C18, as such:
```
#if ((__STDC_VERSION__  == 199901L) || (__STDC_VERSION__ == 201112L) || (__STDC_VERSION__ == 201710L))
/*C99 compiler or C11 or C18*/
```

**Comment by rafilho** [comment]
This was addressed as part of PR #1140.

**Comment by az-iot-builder-01** [comment]
@gerardoti, @rafilho, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
